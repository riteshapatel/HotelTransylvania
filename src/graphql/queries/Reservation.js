/**
 * reservation query
 * @author ritesh.patel
 * App landing screen
 */
import gql from 'graphql-tag';

const Reservation = gql`
query Reservation (
        $where: ReservationWhereUniqueInput!
    ){
        reservation(
            where: $where
        ) {
            id
            name 
            hotelName 
            arrivalDate 
            departureDate 
        }
    
    }
`
export default Reservation;