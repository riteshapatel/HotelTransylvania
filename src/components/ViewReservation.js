/**
 * @class View Reservation component 
 * @author ritesh.patel
 * App landing screen
 */
import React, { Component } from 'react';
import { Text, View, Image } from 'react-native';
import Reservation from '../graphql/queries/Reservation.js';
import { Query, graphql } from 'react-apollo';
import styles from '../styles/Global.style.js';

class ViewReservation extends Component {
    render () { 
        const {navigate} = this.props.navigation;
        const id = this.props.navigation.getParam('id', '');

        return (
            <View style={styles.container}>
                <Text style={styles.coffin_slogan}>Are these monsters gonna kill me?{'\n'}</Text>

                <Query
                    query={Reservation} variables={{where: {id: id}}}>
                    {({ loading, error, data }) => {
                        if (loading) return (<View><Text>Loading...</Text></View>)
                        if (error) return (<View><Text>Error...{error}</Text></View>)
    
                        return (
                            <View style={styles.card}>
                                <Image source={require('../assets/mummy_small.png')} />
                                <View style={styles.coffin_item}>
                                    <Text style={styles.view_coffin}>{data.reservation.name} @ {data.reservation.hotelName}</Text>
                                    <Text style={styles.coffin_date}>({data.reservation.arrivalDate} - {data.reservation.departureDate}){'\n'}</Text>
                                </View>
                            </View>
                        )
                    }}
                </Query>
            </View>
        )
    }
}

export default ViewReservation;