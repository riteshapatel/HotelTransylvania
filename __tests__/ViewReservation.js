/**
 * Unit test for ViewReservation component 
 * @author ritesh.patel
 */
import React from 'react';
import ViewReservation from '../src/components/ViewReservation';
import {MockedProvider} from 'react-apollo/test-utils';
import Reservation from '../src/graphql/queries/Reservation';
import renderer from 'react-test-renderer';

test ('snapshot ViewReservation component', () => {
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const mocks = [
        {
            request: {
                query: Reservation,
                variables: {
                    where: {
                        id: 230729832079907
                    }
                }
            },
            result: {
                data: {
                    reservation: {
                        id: '',
                        name: '',
                        hotelName: '',
                        arrivalDate: '',
                        departureDate: ''
                    }
                },
                loading: {},
                error: {}
            }
        }
    ]    
    const tree = renderer.create(<MockedProvider mocks={mocks}><ViewReservation navigation={navigation} /></MockedProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});