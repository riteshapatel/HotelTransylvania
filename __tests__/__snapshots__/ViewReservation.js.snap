// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot ViewReservation component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C800CB",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Baskerville",
        "fontSize": 24,
        "fontStyle": "italic",
      }
    }
  >
    Are these monsters gonna kill me?
    

  </Text>
  <View>
    <Text>
      Loading...
    </Text>
  </View>
</View>
`;
